tests/e2e/logging/test_openshift-logging.py

```diff
@@ -72,44 +72,82 @@ def finalizer():
         helpers.wait_for_resource_state(resource=pod_obj, state=constants.STATUS_RUNNING)
         return pod_obj, pvc_obj
 
-    @retry(ModuleNotFoundError, tries=5, delay=200, backoff=1)
-    def validate_project_exists(self, pvc_obj):
+    @retry(ModuleNotFoundError, tries=5, delay=400, backoff=1)
+    def validate_project_exists(self, project):
         """
         This function checks whether the new project exists in the
         EFK stack
+
+        Args:
+            project (str): The project
+
         """
-        pod_list = get_all_pods(namespace='openshift-logging')
-        elasticsearch_pod = [
-            pod.name for pod in pod_list if pod.name.startswith('elasticsearch')
-        ]
-        elasticsearch_pod_obj = get_pod_obj(
-            name=elasticsearch_pod[1], namespace='openshift-logging'
-        )
-        project_index = elasticsearch_pod_obj.exec_cmd_on_pod(
-            command='indices', out_yaml_format=False
-        )
-        project = pvc_obj.project.namespace
 
-        if project in project_index:
-            logger.info(f'The project {project} exists in the EFK stack')
-            for item in project_index.split("\n"):
-                if project in item:
-                    logger.info(item.strip())
-                    assert 'green' in item.strip(), f"Project {project} is Unhealthy"
+        elasticsearch_pod_obj = self.get_elasticsearch_pod_obj()
+        if get_ocp_version() <= "4.4":
+
+            project_index = elasticsearch_pod_obj.exec_cmd_on_pod(
+                command='indices', out_yaml_format=False
+            )
+            if project in project_index:
+                logger.info(f'The project {project} exists in the EFK stack')
+                for item in project_index.split("\n"):
+                    if project in item:
+                        logger.info(item.strip())
+                        assert 'green' in item.strip(), f"Project {project} is Unhealthy"
+            else:
+                raise ModuleNotFoundError
         else:
-            raise ModuleNotFoundError
+            cmd = (
+                'es_util --query=*/_search?pretty -d \'{"query": {"match":'
+                f' {{"kubernetes.namespace_name": "{project}"}}}}}}\''
+            )
+            project_out = elasticsearch_pod_obj.exec_cmd_on_pod(
+                command=cmd, out_yaml_format=False
+            )
+            if project_out:
```

Comment:
in which case `project_out` can be empty?

Reply:
Changed